unstructured/cleaners/core.py

```diff
@@ -108,6 +107,87 @@ def group_broken_paragraphs(
     return "\n\n".join(clean_paragraphs)
 
 
+def new_line_grouper(
+    text: str,
+    paragraph_split: re.Pattern = LINE_BREAK_RE,
+) -> str:
+    """
+    Concatenates text document that has one-line paragraph break pattern
+
+    For example,
+
+    Iwan Roberts
+    Roberts celebrating after scoring a goal for Norwich City
+    in 2004
+
+    Will be returned as:
+
+    Iwan Roberts\n\nRoberts celebrating after scoring a goal for Norwich City\n\nin 2004
+    """
+    paragraphs = paragraph_split.split(text)
+    clean_paragraphs = []
+    for paragraph in paragraphs:
+        if not paragraph.strip():
+            continue
+        clean_paragraphs.append(paragraph)
+    return "\n\n".join(clean_paragraphs)
+
+
+def blank_line_grouper(
+    text: str,
+) -> str:
+    """
+    Concatenates text document that has blank-line paragraph break pattern
+
+    For example,
+
+    "Vestibulum auctor dapibus neque.
+
+    Nunc dignissim risus id metus."
+
+    Will be returned as:
+
+    "Vestibulum auctor dapibus neque.\n\nNunc dignissim risus id metus.\n\n"
+
+    The function is now calling group_broken_paragraphs function.
+    """
+    return group_broken_paragraphs(text)
+
+
+def auto_paragraph_grouper(
+    text: str,
+    line_split: re.Pattern = LINE_BREAK_RE,
+    max_line_count: int = 2000,
+    threshold: float = 0.5,
```

Comment:
intuitively, it feels this could br lower. E.g., if there is one blank line every 10 lines, it probably means splitting on the blank lines is the better approach than by newlines.

Reply:
Fixed and pushed.